src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx

```diff
@@ -32,6 +41,35 @@ const EditSamplesConfirmationModal = ({
   onClose,
   open,
 }: Props): JSX.Element | null => {
+  const [isValid, setIsValid] = useState(false);
+  const [metadata, setMetadata] =
+    useState<SampleIdToEditMetadataWebform | null>(null);
+
+  const { data: namedLocationsData } = useNamedLocations();
+  const namedLocations =
+    namedLocationsData?.namedLocations || ([] as NamedGisaidLocation[]);
+
+  function structureInitialMetadata(item: Sample): SampleEditMetadataWebform {
+    const i = _.pick(
+      item,
+      Object.keys(SAMPLE_EDIT_WEBFORM_METADATA_KEYS_TO_HEADERS)
+    ) as SampleEditMetadataWebform;
+    const cl = i.collectionLocation;
+    if (cl) {
+      cl.name = `${cl.region}/${cl.country}/${cl.division}/${cl.location}`;
+    }
+    return i;
+  }
+
+  useMemo(() => {
+    const structuredMetadata: SampleIdToEditMetadataWebform = {};
+    checkedSamples.forEach((item) => {
+      structuredMetadata[item.privateId as keyof typeof structuredMetadata] =
```

Comment:
```suggestion
      structuredMetadata[item.privateId] =
```

Reply:
for some reason the compiler wasn't liking that approach, but i can try again and see